src/main/java/frc/robot/Drive.java

```diff
@@ -0,0 +1,212 @@
+    double COUNTS_PER_INCH = (COUNTS_PER_ROTATION * DRIVE_GEAR_REDUCTION) / WHEEL_CIRCUMFERENCE;
+    CANSparkMax leftFront = new CANSparkMax(12, MotorType.kBrushless);
+    CANSparkMax rightFront = new CANSparkMax(10, MotorType.kBrushless);
+    CANSparkMax leftBack = new CANSparkMax(13, MotorType.kBrushless);
+    CANSparkMax rightBack = new CANSparkMax(11, MotorType.kBrushless);
+    RelativeEncoder lFrontEncoder = leftFront.getEncoder(SparkMaxRelativeEncoder.Type.kHallSensor, COUNTS_PER_ROTATION);
+    RelativeEncoder rFrontEncoder = rightFront.getEncoder(SparkMaxRelativeEncoder.Type.kHallSensor, COUNTS_PER_ROTATION);
+    RelativeEncoder lBackEncoder = leftBack.getEncoder(SparkMaxRelativeEncoder.Type.kHallSensor, COUNTS_PER_ROTATION);
+    RelativeEncoder rBackEncoder = rightBack.getEncoder(SparkMaxRelativeEncoder.Type.kHallSensor, COUNTS_PER_ROTATION);
+    boolean eDriveDone = false;
+    boolean alignDone = false;
+
+    // Gyro declrations and initializations
+    ADXRS450_Gyro gyro = new ADXRS450_Gyro();
+    Encoder leftEncoder = new Encoder(1, 2, false);
+    Encoder rightEncoder = new Encoder(3, 4, false);
+    boolean turnDrive = false;
+    double currentAngle;
+
+    // Limelight 
+    Apriltags aprilTags = new Apriltags();
+
+  public Drive() {
+    // rFrontEncoder.setInverted(true);
+    rightFront.setInverted(true);
+    // rBackEncoder.setInverted(true);
+    rightBack.setInverted(true);
+  }
+
+  public void robotInit() {
+
+  }
+
+    public void encoderDrive(double speed, double distance, String direction, double timeOut) {
+        int motorTarget = (int)(distance * COUNTS_PER_INCH);
+        eDriveDone = false;
+        
+        double timeStarted = System.currentTimeMillis();
+    
+        lFrontEncoder.setPosition(0);
+        rFrontEncoder.setPosition(0);
+        lBackEncoder.setPosition(0);
+        rBackEncoder.setPosition(0);
+    
+        if (direction == "forward") {
+          leftFront.set(speed);
+          rightFront.set(speed);
+          leftBack.set(speed);
+          rightBack.set(speed);  
+        }
+        else if (direction == "backward") {
+          leftFront.set(-speed);
+          rightFront.set(-speed);
+          leftBack.set(-speed);
+          rightBack.set(-speed);
+        }
+        
+        if(Math.abs(lFrontEncoder.getPosition()) >= motorTarget || Math.abs(rFrontEncoder.getPosition()) >= motorTarget || Math.abs(lBackEncoder.getPosition()) >= motorTarget || Math.abs(rBackEncoder.getPosition()) >= motorTarget || System.currentTimeMillis() > (timeStarted + timeOut)) {
+          leftFront.set(0);
+          rightFront.set(0);
+          leftBack.set(0);
+          rightBack.set(0);
+          eDriveDone = true;
+        }
+
+        double allEncoder = lFrontEncoder.getPosition() + rFrontEncoder.getPosition();
+    
+      }
+
+    //This method is called on during autonomous for turning
+    public void turnDrive(int timeOut, double turnDegrees, double speed) {
```

Comment:
I'd recommend matching the syntax of encoder drive as much as possible here

```suggestion
    public void turnDrive(double speed, double turnDegrees, int timeOut) {
```